src/cli/persist.go

```diff
@@ -0,0 +1,31 @@
+package cli
+
+import (
+	"github.com/spf13/cobra"
+)
+
+const persistShortDescription = `Persist created and deleted configs/rows in manifest`
+const persistLongDescription = `Command "persist"
+
+This command writes the changes from the filesystem to the manifest.
+- If you have created a new config/row, this command will write record to the manifest with a unique ID.
+- If you have deleted a config/row, this command will delete record from the manifest.
+
```

Comment:
citam ze persist command je len pre vytvorene a zmazane configs/rows. Upravy config rows(cez api) sa tu nezohladnuju? tj zmenim config cez api, zavolam persist a manifest sa neupravi alebo upravi?

Reply:
1. ak zmenis config cez API, tak zavolas `pull` a zmena sa ti prejavi aj lokalne .... 
2. ak zmenis config lokalne a chces ho mat v projekte, zavolas `push`
3. manifest obsahuje iba zmienku, kde je config ulozeny na disku, ... takze v tomto pripade "zmenim" nedava zmysel, ... manifest sa upravuje iba ked vytvoris/zmazes
4. ak vytvorim config/row manualne, tak o nom CLI tool nevie a preto sa musi zavolat `persist` aby sa pridal do manifestu.
5. podobne ako ked zmazem config/row manualne.
6. mozeme zavolat ak je to stale nejasne :)
7. mozeme dat nejaku lepsiu `help`, ak je tato nedostatocna, ... mne s ostatnymi prikazmi dava zmysel, ale tvoj pohlad je asi menej zaujaty .... bezny uzivatel, zacne s prikazmi `init`, `pull`, `push` ... a az potom `persist`, takze skus pozriet help v tomto poradi.

`manifest.json`, vyzera napr. takto:
```
{
  "version": 1,
  "project": {
    "id": 8614,
    "apiHost": "connection.keboola.com"
  },
  "sortBy": "path",
  "naming": {
    "branch": "{branch_name}",
    "config": "{component_type}/{component_id}/{config_name}",
    "configRow": "rows/{config_row_name}"
  },
  "branches": [
    {
      "id": 23613,
      "path": "bar"
    },
    {
      "id": 23612,
      "path": "foo"
    },
    {
      "id": 15159,
      "path": "main"
    }
  ],
  "configurations": [
    {
      "branchId": 23613,
      "componentId": "ex-generic-v2",
      "id": "728967753",
      "path": "extractor/ex-generic-v2/empty",
      "rows": []
    },
    {
      "branchId": 23613,
      "componentId": "keboola.ex-db-mysql",
      "id": "728967771",
      "path": "extractor/keboola.ex-db-mysql/with-rows",
      "rows": [
        {
          "id": "728967775",
          "path": "rows/disabled"
        },
        {
          "id": "728967776",
          "path": "rows/test-view"
        },
        {
          "id": "728967777",
          "path": "rows/users"
        }
      ]
    },
    {
      "branchId": 23613,
      "componentId": "keboola.python-transformation-v2",
      "id": "728967772",
      "path": "transformation/keboola.python-transformation-v2/python-transformation",
      "rows": []
    },
    {
      "branchId": 23613,
      "componentId": "keboola.snowflake-transformation",
      "id": "728967773",
      "path": "transformation/keboola.snowflake-transformation/snowflake-transformation",
      "rows": []
    },
    {
      "branchId": 23612,
      "componentId": "ex-generic-v2",
      "id": "728967753",
      "path": "extractor/ex-generic-v2/empty",
      "rows": []
    },
    {
      "branchId": 23612,
      "componentId": "ex-generic-v2",
      "id": "728967769",
      "path": "extractor/ex-generic-v2/without-rows",
      "rows": []
    },
    {
      "branchId": 23612,
      "componentId": "transformation",
      "id": "728967770",
      "path": "other/transformation/old-transformation",
      "rows": [
        {
          "id": "728967774",
          "path": "rows/old-transformation-snfk"
        }
      ]
    },
    {
      "branchId": 15159,
      "componentId": "ex-generic-v2",
      "id": "728967753",
      "path": "extractor/ex-generic-v2/empty",
      "rows": []
    }
  ]
}

```

Reply:
Mozno by som tam mal pridat na koniec:
`If you want the changes to take effect in the remote project, run `push` after the` persist` command.`

Reply:
Je mozne ze to vnimam zle a nieco mi tam unika no ma este napada ci ten persist command neni zbytocny a teda ze by ho mohol kompletne zastupit pull command
- 1. vytvorim config/row manualne a o nom CLI tool nevie - nepomohol by tu pull command teda ze zavola pull a vytvori sa riadok v manifest podobne ako to robi persist?
- 2. zmazem config/row manualne - rovnaku by tu nepomohol pull command teda ze by zmazal aj z manifestu?

Asi by tam mohlo nejaka hlaska + yes/no otazka ze sa remote config zmenil(pribudol/odobral config) ci to chce pullnut takto "natvrdo".

Reply:
asi by bolo lepsie si zavolat:

- `pull` - nemoze nahradit `persist` ... `pull` synchronizuje remote -> local, ... nema to nic spolocne s tym, ze ty si lokalne vytvoris novy config.
- Mozno si myslel `push`, tj. sync local -> remote, ... tam by to slo pridat ale je to naschval oddelene.
  - Je to preto, lebo tento tool moze naraz pouzivat viacero uzivatelov spolu s Gitom.
  - Use case je, ze pracuju nezavisle napr. 3 uzivatelia s CLI nastrojom, vo svojej GIT vetve.
  - Kazdy si vytvori novy config, zavola `persist` a config/row sa priradia unikatne ID, ALE! remote stav sa nezmenil.
  - Vsetci dokoncia svoju pracu, poslu PR, ... mergne sa to do master, ... mame 3 nove konfiguracie, ktore su v Gite, ale este neboli syncnute, nezavolal sa `push`.
  - CIcko, ktore sa vola nad kazdou zmenou v `master` vetve zavola `push` command, a 3 nove konfiguracie su syncnute do projektu
  - ak by sme spojili `persist` a `push`, tak tento use-case nie je mozny. Lebo hned ako by si vytvoril config/row, tak by si ho musel pushnut -> nemohol by si ho poslat na review.
  - ak by si ho `nepushol`, tak by zase nemal pridelene ID, .... CIcko by zavolalo `push`,.... ale nove IDcka, ktore vznikli by sa uz nedostali naspat do Gitu a k uzivatelom co to vytvorili.
  
Asi bude najlepsi call, ... presne tieto use-cases sme riesili s produktom a najma s Chochom na calle, a vznikol z toho ten `persist` command.

Reply:
ked si budes chciet zavolat, tak mi napis DM na slacku ;)